src/service/cache-cid-api.js

```diff
@@ -0,0 +1,209 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * amp-lUSTQLTz0csodkt327ZRHm30PG-6sJOuZwTh-PPmzqRMGZrTCRBMJ8Pfdw8f_UGC
+ * amp-lUSTQLTz0csodkt327ZRHm30PG-6sJOuZwTh-PPmzqRMGZrTCRBMJ8Pfdw8f_UGC
+ */
+
+import {Services} from '../services';
+import {dev, user} from '../log';
+import {dict} from '../utils/object';
+import {getSourceOrigin} from '../url';
+
+const GOOGLE_CLIENT_ID_API_META_NAME = 'amp-google-client-id-api';
+const CID_API_SCOPE_WHITELIST = {
+  'googleanalytics': 'AMP_ECID_GOOGLE',
+};
+const API_KEYS = {
+  'googleanalytics': 'AIzaSyA65lEHUEizIsNtlbNo-l2K18dT680nsaM',
+};
+
+/**
+ * The Client ID service key.
+ * @const @private {string}
+ */
+const SERVICE_KEY_ = 'AIzaSyDKtqGxnoeIqVM33Uf7hRSa3GJxuzR7mLc';
+
+const TAG = 'CacheCidApi';
+const CACHE_API_URL = 'https://ampcid.google.com/v1/cache:getClientId?key=';
+
+const TIMEOUT = 30000;
```

Comment:
Please add JSDoc for all of these constants.

Reply:
Done.